contracts/Tokenator.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.24;
+
+import {ERC20, ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract Tokenator is ERC20Burnable, Ownable {
+    using SafeERC20 for IERC20;
+
+    IERC20 underlyingToken;
+    uint256 private constant PRECISION = 1e18;
+    uint256 public immutable commencement;
+    uint256 public immutable durationSeconds;
+
+    error ZeroAddress();
+    error CommencementPassed();
+    error CommencementNotPassed();
+    error CannotRescueUnderlyingToken();
+
+    event Deployed(
+        address deployer,
+        IERC20 underlyingToken,
+        uint256 commencement,
+        uint256 durationDays
+    );
+    event ClaimUnderlyingToken(
+        address indexed user,
+        address indexed to,
+        uint256 tokenatorAmount,
+        uint256 underlyingTokenAmount
+    );
+    event EmergencyWithdrawal(uint256 amount, address indexed to);
+    event TokensRescued(IERC20 token, uint256 amount, address indexed to);
+
+    /// @notice following the airdrop, the deployer is required to burn any surplus Tokenator
```

Comment:
there will be no airdrop, so the comment should be updated. But the deployer is responsible for sending underlying tokens in

Reply:
done